app/resources/v1/users.js

```diff
@@ -242,8 +248,39 @@ exports.get = async function (req, res) {
     }
   }
 
-  // TODO this function seems like it could be replaced by
-  // sequelize findByPK
+  const BTPTokenCheck = async function (userData) {
+    // check for coil provider to set access token to stream payments
+    const coilData = userData.identities.find(
+      (item) => item.provider === 'coil'
+    )
+    if (coilData !== undefined) {
+      // fetch and return btpToken
+      let btpToken = await getBTPToken(coilData.access_token)
+      if (typeof btpToken === 'undefined') {
+        // token may be expired, so lets refresh and try
+        const newAccessToken = await refreshAccessToken(coilData.refresh_token)
+
+        const identities = userData.identities
+        coilData.access_token = newAccessToken
+        addOrUpdateByProviderName(identities, coilData)
+        await User.update(
+          {
+            identities: identities
+          },
+          { where: { auth0Id: userData.auth0Id }, returning: true }
+        )
+        // return btp token after updating access token
+        btpToken = await getBTPToken(newAccessToken)
+      }
+
+      // express-sesion seems to indicate this line is all that would be needed to add to cookies
+      req.session.btpToken = btpToken
```

Comment:
I don't think we have the express-session package installed. If you remove this line, and only use `res.cookie()` will you still get the btpToken client side?